include/ppx/knob.h

```diff
@@ -314,6 +317,75 @@ class KnobDropdown final
     std::vector<T> mChoices;
 };
 
+// KnobFlag is intended for parameters that cannot be adjusted when the application is run
+// They will not be displayed in the UI and cannot have their values changed by the application
+// Their values are the default unless otherwise set through commandline flags on application start up
+template <typename T>
+class KnobFlag final
+    : public Knob
+{
+public:
+    KnobFlag(const std::string& flagName, T defaultValue)
+        : Knob(flagName, false)
+    {
+        SetValue(defaultValue);
+    }
+
+    KnobFlag(const std::string& flagName, T defaultValue, T minValue, T maxValue)
+        : Knob(flagName, false)
+    {
+        static_assert(std::is_arithmetic_v<T>, "KnobFlag can only be defined with min/max when it's of arithmetic type");
+        PPX_ASSERT_MSG(minValue < maxValue, "invalid range to initialize KnobFlag");
+        PPX_ASSERT_MSG(minValue <= defaultValue && defaultValue <= maxValue, "defaultValue is out of range");
+
+        SetValidator([minValue, maxValue](T newValue) {
+            if (newValue < minValue || newValue > maxValue) {
+                return false;
+            }
+            return true;
+        });
+
+        SetValue(defaultValue);
+    }
+
+    T GetValue() const { return mValue; }
+
+    void SetValidator(std::function<bool(T)> validatorFunc) { mValidatorFunc = validatorFunc; }
+
+private:
+    void Draw() override
+    {
+        std::stringstream ss;
```

Comment:
Related to the other comment, but this could be used by knob framework users by setting SetVisible(true). If that's intended we can leave it, if this is just for debugging while developing knobs perhaps we should remove it.